src/libexpr/primops/fetchGit.cc

```diff
@@ -225,17 +225,28 @@ static void prim_fetchGit(EvalState & state, const Pos & pos, Value * * args, Va
     // whitelist. Ah well.
     state.checkURI(url);
 
-    auto gitInfo = exportGit(state.store, url, ref, rev, name);
-
-    state.mkAttrs(v, 8);
-    mkString(*state.allocAttr(v, state.sOutPath), gitInfo.storePath, PathSet({gitInfo.storePath}));
-    mkString(*state.allocAttr(v, state.symbols.create("rev")), gitInfo.rev);
-    mkString(*state.allocAttr(v, state.symbols.create("shortRev")), gitInfo.shortRev);
-    mkInt(*state.allocAttr(v, state.symbols.create("revCount")), gitInfo.revCount);
-    v.attrs->sort();
-
-    if (state.allowedPaths)
-        state.allowedPaths->insert(state.store->toRealPath(gitInfo.storePath));
+    try {
+        auto gitInfo = exportGit(state.store, url, ref, rev, name);
+
+        state.mkAttrs(v, 8);
+        mkString(*state.allocAttr(v, state.sOutPath), gitInfo.storePath, PathSet({gitInfo.storePath}));
+        mkString(*state.allocAttr(v, state.symbols.create("rev")), gitInfo.rev);
+        mkString(*state.allocAttr(v, state.symbols.create("shortRev")), gitInfo.shortRev);
+        mkInt(*state.allocAttr(v, state.symbols.create("revCount")), gitInfo.revCount);
+        v.attrs->sort();
+
+        if (state.allowedPaths)
+            state.allowedPaths->insert(state.store->toRealPath(gitInfo.storePath));
+    } catch (ExecError & e) {
```

Comment:
It's better to put this check in `exportGit` because (on the flakes branch) it's used in other places as well.

Reply:
Sure, should I do the same for mercurial?